nginx/files/proxy.conf

```diff
@@ -44,6 +44,10 @@ server {
       send_timeout 600;
       {%- endif %}
 
+      {%- if site.proxy.rewrite is defined %}
```

Comment:
This has much more potential to add as a cool new feature. Rewrite module can do much more, example from docs:
```
if ($http_user_agent ~ MSIE) {
    rewrite ^(.*)$ /msie/$1 break;
}

if ($http_cookie ~* "id=([^;]+)(?:;|$)") {
    set $id $1;
}

if ($request_method = POST) {
    return 405;
}

if ($slow) {
    limit_rate 10k;
}

if ($invalid_referer) {
    return 403;
}
```

I can imagine to have:
```
rewrites:
  - condition: "$request_method == POST"
    action: "return 405"
  - condition "$http_user_agent ~ MSIE"
    action: "rewrite ^(.*)$ /msie/$1 break"
```